common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ErrorHandler.java

```diff
@@ -87,9 +87,11 @@ public boolean shouldRetryError(Throwable t) {
         return false;
       }
 
-      // If the block is too late or stale block push, there is no need to retry it
+      // If the block is too late or the invalid block push or the attempt is not the latest one,
+      // there is no need to retry it
       return !(t instanceof BlockPushNonFatalFailure &&
         (((BlockPushNonFatalFailure) t).getReturnCode() == TOO_LATE_BLOCK_PUSH ||
+          ((BlockPushNonFatalFailure) t).getReturnCode() == TOO_OLD_ATTEMPT_PUSH ||
           ((BlockPushNonFatalFailure) t).getReturnCode() == STALE_BLOCK_PUSH));
```

Comment:
Shall we make this a static method of `ReturnCode`?

Reply:
Added static method to get the should not retry error code in latest PR.